operator/src/main/java/oracle/kubernetes/operator/ConversionWebhookMain.java

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2022, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.operator;
+
+import java.util.Properties;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import oracle.kubernetes.operator.calls.CallResponse;
+import oracle.kubernetes.operator.helpers.CallBuilder;
+import oracle.kubernetes.operator.helpers.CrdHelper;
+import oracle.kubernetes.operator.logging.MessageKeys;
+import oracle.kubernetes.operator.rest.RestConfigImpl;
+import oracle.kubernetes.operator.rest.WebhookRestServer;
+import oracle.kubernetes.operator.steps.DefaultResponseStep;
+import oracle.kubernetes.operator.steps.InitializeWebhookIdentityStep;
+import oracle.kubernetes.operator.utils.Certificates;
+import oracle.kubernetes.operator.work.NextAction;
+import oracle.kubernetes.operator.work.Packet;
+import oracle.kubernetes.operator.work.Step;
+import oracle.kubernetes.weblogic.domain.model.DomainList;
+
+import static oracle.kubernetes.operator.helpers.NamespaceHelper.getWebhookNamespace;
+
+/** A Domain Custom Resource Conversion Webhook for WebLogic Kubernetes Operator. */
+public class ConversionWebhookMain extends BaseMain {
+  private boolean warnedOfCrdAbsence;
+  private static final NextStepFactory NEXT_STEP_FACTORY = ConversionWebhookMain::createInitializeWebhookIdentityStep;
+
+  static class ConversionWebhookMainDelegateImpl extends CoreDelegateImpl {
+    public ConversionWebhookMainDelegateImpl(Properties buildProps, ScheduledExecutorService scheduledExecutorService) {
+      super(buildProps, scheduledExecutorService);
+    }
+
+    private void logStartup() {
+      ConversionWebhookMain.LOGGER.info(MessageKeys.CONVERSION_WEBHOOK_STARTED, buildVersion,
+              deploymentImpl, deploymentBuildTime);
+      ConversionWebhookMain.LOGGER.info(MessageKeys.WEBHOOK_CONFIG_NAMESPACE, getWebhookNamespace());
+    }
+  }
+
+  /**
+   * Entry point.
+   *
+   * @param args none, ignored
+   */
+  public static void main(String[] args) {
+    ConversionWebhookMain main = createMain(getBuildProperties());
+
+    try {
+      main.startDeployment(main::completeBegin);
+
+      // now we just wait until the pod is terminated
+      main.waitForDeath();
+
+      // stop the webhook REST server
+      stopWebhookRestServer();
+    } finally {
```

Comment:
What happened if it crashed?  Does k8s automatically restart it and how will the user know something is wrong?

Reply:
Yes, k8s will restart. We are making sure that the webhook can support replicas > 1 so that there isn't a single point of failure if one webhook crashes.

Reply:
Created JIRA OWLS-97471 to track the event generation for conversion webhook unhappy paths.